stable/mongodb-replicaset/init/on-start.sh

```
@@ -151,67 +151,71 @@ fi
 
 init_mongod_standalone
 
-log "Peers: ${peers[*]}"
-log "Starting a MongoDB replica"
-mongod --config /data/configdb/mongod.conf --dbpath=/data/db --replSet="$replica_set" --port="${port}" "${auth_args[@]}" --bind_ip=0.0.0.0 2>&1 | tee -a /work-dir/log.txt 1>&2 &
-pid=$!
-trap shutdown_mongo EXIT
-
-log "Waiting for MongoDB to be ready..."
-retry_until "localhost" "db.adminCommand('ping').ok" "1"
-log "Initialized."
-
-# try to find a master
-for peer in "${peers[@]}"; do
-    log "Checking if ${peer} is primary"
-    # Check rs.status() first since it could be in primary catch up mode which db.isMaster() doesn't show
-    if [[ $(mongo admin --host "${peer}" "${admin_creds[@]}" "${ssl_args[@]}" --quiet --eval "rs.status().myState") == "1" ]]; then
-        retry_until "${peer}" "db.isMaster().ismaster" "true"
-        log "Found primary: ${peer}"
-        primary="${peer}"
-        break
-    fi
-done
-
-if [[ "${primary}" = "${service_name}" ]]; then
-    log "This replica is already PRIMARY"
-elif [[ -n "${primary}" ]]; then
-    if [[ $(mongo admin --host "${primary}" "${admin_creds[@]}" "${ssl_args[@]}" --quiet --eval "rs.conf().members.findIndex(m => m.host == '${service_name}:${port}')") == "-1" ]]; then
-      log "Adding myself (${service_name}) to replica set..."
-      if (mongo admin --host "${primary}" "${admin_creds[@]}" "${ssl_args[@]}" --eval "rs.add('${service_name}')" | grep 'Quorum check failed'); then
-          log 'Quorum check failed, unable to join replicaset. Exiting prematurely.'
-          exit 1
+if [[ "${DO_NOT_INIT}" == "true" ]]; then
+    log "Skipping initialization"
+else
```

Don't add an else, just fail fast by logging and exit 0.